File: boat-spring-boot-sample/src/test/java/sample/java/xyz/srclab/spring/boot/task/AsyncService.java
package sample.java.xyz.srclab.spring.boot.task;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.scheduling.annotation.Async;
import org.springframework.stereotype.Service;
import org.testng.Assert;
import xyz.srclab.common.base.Current;

@Service
public class AsyncService {

    private static final Logger logger = LoggerFactory.getLogger(AsyncService.class);

    @Async
    public void testAsync() {
        logger.info("Thread: {}", Current.thread().getName());
        Assert.assertTrue(Current.thread().getName().startsWith("666"));
    }
}

File: boat-spring-boot-sample/src/test/kotlin/sample/kotlin/xyz/srclab/spring/boot/task/AsyncService.kt
package sample.kotlin.xyz.srclab.spring.boot.task

import org.slf4j.LoggerFactory
import org.springframework.scheduling.annotation.Async
import org.springframework.stereotype.Service
import org.testng.Assert
import xyz.srclab.common.lang.Current

@Service
open class AsyncService {

    @Async
    open fun testAsync() {
        logger.info("Thread: {}", Current.thread.name)
        Assert.assertTrue(Current.thread.name.startsWith("666"))
    }

    companion object {
        private val logger = LoggerFactory.getLogger(AsyncService::class.java)
    }
}